dot/state/epoch.go

```diff
@@ -56,7 +72,7 @@ type GenesisEpochDescriptor struct {
 
 // EpochState tracks information related to each epoch
 type EpochState struct {
-	db           GetterPutterNewBatcher
+	db           GetterPutterNewBatcherPrefixIter
```

Comment:
I thought about it, but then I saw that we create a new table out of `database.Database` with a epochprefix attached to it 
like this  
`&EpochState{
		baseState:      baseState,
		blockState:     blockState,
		db:             database.NewTable(db, epochPrefix),`
changing it to database.database will implied  a major change in how we interact with this interface.
can you give me more context of why we would want to use `database.Database` instead of the table impl?

Reply:
database.Table then?